### PR TITLE
Add DMN 1.5 boxed expressions documentation

### PR DESCRIPTION
**Pull request by jomarko**

### CODE REVIEW

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 1]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+
+NOTE: This feature was introduced as part of {PRODUCT} 9.1 and DMN 1.5 releases.
+
+[id="con-dmn-for-expressions"]
+==== Boxed iterator expressions
+
+[role="_abstract"]
+A boxed iterator offers a visual representation of an iterator statement.
+For the `for` loop, the right part of the `for` displays the iterator variable name.
+The second row holds an expression representing the collection that will be iterated over.
```
Comment (kbowers-ibm) [inline runtime-services-modeling/decisions-with-dmn.adoc:691]:
```suggestion
The second row must contain an expression that resolves to a collection, which will be iterated over.
```

Comment (kbowers-ibm) [reply runtime-services-modeling/decisions-with-dmn.adoc:691]: Or 
The `in` row must contain an expression that resolves to a collection, which will be iterated over. 
Just personal preference here, but at first glance it sounded like the second row and the in row were different rows.

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 2]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+The expression in the `in` row must resolve to a collection.
```
Comment (kbowers-ibm) [inline runtime-services-modeling/decisions-with-dmn.adoc:692]:
```suggestion
```

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 3]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+image::runtime-services-modeling/dmn-some-expression.png[]
+
+NOTE: This feature was introduced as part of {PRODUCT} 9.1 and DMN 1.5 releases.
+
+[id="con-dmn-filter-expressions"]
+==== Boxed filter expressions
+
+[role="_abstract"]
+A boxed filter offers a visual representation of collection filtering.
+The top part is an expression that is the collection to be filtered.
```
Comment (kbowers-ibm) [inline runtime-services-modeling/decisions-with-dmn.adoc:739]:
```suggestion
The top part contains an expression representing the collection to be filtered.
```

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 4]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+The bottom part, between the square brackets, holds the filter expression.
+
+The example below is an alternative way of defining logic as <<flight-capacity-example>>.
```
Comment (kbowers-ibm) [inline runtime-services-modeling/decisions-with-dmn.adoc:742]:
```suggestion
The example below is an alternative way of defining logic similar to the <<flight-capacity-example>> example.
```

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 5]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+
+[role="_abstract"]
+A boxed iterator offers a visual representation of an iterator statement.
+For the `every` loop, the right part of the `every` displays the iterator variable name.
+The second row holds an expression representing the collection that will be iterated over.
+The expression in the `in` row MUST resolve to a collection.
+The last line is an expression that will be evaluated on each item.
+The expression defined in the `satisfies` must resolve to a boolean.
+
+.Every expression of Will Flight Together decision
```
Comment (kbowers-ibm) [inline runtime-services-modeling/decisions-with-dmn.adoc:713]: Probably not a big deal, but should be "Will Fly Together" / "Will Not Fly Together".

**Comment by jomarko** [comment]
@kbowers-ibm thank you, I haven't applied all suggestions, because some text is a copy of the text from the editor ui. I think if we want to address remaining points, we should update also the editor ui. what do you think?

image

**Review by kbowers-ibm — APPROVED** [review]
Yeah fair enough; I think you're right and it's best to keep them consistent. The change was more just for style/clarity as anyone looking will see that the "in" row is the 2nd row, so I think it's fine to remain as is. Thanks Jozef!

**Review by danielzhe — APPROVED** [review]
Thank you! Good examples you did for the new expressions!

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 6]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+.Iterator expression of Approved credit score agencies
+image::runtime-services-modeling/dmn-iterator-expression.png[]
+
+NOTE: This feature was introduced as part of the {PRODUCT} 9.1 and DMN 1.5 releases.
+
+[id="con-dmn-every-expressions"]
+==== Boxed every expressions
+
+[role="_abstract"]
+A boxed iterator offers a visual representation of an iterator statement.
```
Comment (ljmotta) [inline runtime-services-modeling/decisions-with-dmn.adoc:706]: It has the same text of the `Boxed iterator expressions` section, maybe changing the previous section to `Boxed for expression`? And be more specific on the paragraph as this is related to the `every` expression.

`A boxed every offers a visual representation of an iterator statement`

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 7]
```diff
@@ -653,12 +661,91 @@ image::dmn-list-expression-example.png[]
...
+.Every expression of Will Fly Together decision
+image::runtime-services-modeling/dmn-every-expression.png[]
+
+NOTE: This feature was introduced as part of the {PRODUCT} 9.1 and DMN 1.5 releases.
+
+[id="con-dmn-some-expressions"]
+==== Boxed some expressions
+
+[role="_abstract"]
+A boxed iterator offers a visual representation of an iterator statement.
```
Comment (ljmotta) [inline runtime-services-modeling/decisions-with-dmn.adoc:722]: Same here.

**Review by ljmotta — APPROVED** [review]
@jomarko Thanks a lot for this PR. You've put a lot of effort on the documentation, well done!

File: runtime-services-modeling/decisions-with-dmn.adoc (hunk 6 above)
Comment (jomarko) [reply runtime-services-modeling/decisions-with-dmn.adoc]: done

File: runtime-services-modeling/decisions-with-dmn.adoc (hunk 7 above)
Comment (jomarko) [reply runtime-services-modeling/decisions-with-dmn.adoc]: done

**Comment by jomarko** [comment]
@kbowers-ibm @ljmotta @danielzhe hi, after a look on the [PR](https://github.com/apache/incubator-kie-tools/pull/2296), I decided to add a piece of `filter` documentation using the `item` reserved keyword.
This requires review of one more [PR](https://github.com/bamoe/bamoe-docs-source-code-examples/pull/4) as I would like to have available source code for the documentation readers.

Would you mind to take one more look on #108 and [PR](https://github.com/bamoe/bamoe-docs-source-code-examples/pull/4)?

**Review by LightGuard — APPROVED** [review]

**Comment by LightGuard** [comment]
Please let me know when we all feel good to merge.

File: runtime-services-modeling/decisions-with-dmn.adoc [hunk 8]
```diff
@@ -653,12 +661,101 @@ image::dmn-list-expression-example.png[]
...
+The top part is an expression that is the collection to be filtered.
+The bottom part, between the square brackets, holds the filter expression.
+
+The example below is an alternative way of defining logic as <<flight-capacity-example>>.
+
+.Filter expression of rebooked flights
+image::runtime-services-modeling/dmn-filter-expression.png[]
+
+Furthermore, the `filter` expression brings a reserved keyword `item`.
+Using this keyword you can explicitly access items of the filtered collection.
```
Comment (kbowers-ibm) [inline runtime-services-modeling/decisions-with-dmn.adoc:748]:
```suggestion
Using this keyword you can explicitly access items from the filtered collection.
```

**Review by kbowers-ibm — APPROVED** [review]
LGTM, thanks Jozef! :)

**Review by LightGuard — APPROVED** [review]

**Comment by LightGuard** [comment]
Any other changes expected, or should we merge?

**Comment by jomarko** [comment]
@LightGuard we need to finish https://github.com/bamoe/bamoe-docs-source-code-examples/pull/4 and merge just before merging #108.

**Comment by jomarko** [comment]
@LightGuard hi, the related [PR ](https://github.com/bamoe/bamoe-docs-source-code-examples/pull/4) was merged. I think We are fine to merge #108 now.

**Pull request merged**